src/third_party/wiredtiger/src/evict/evict_lru.c

```diff
@@ -1392,7 +1392,7 @@ __evict_walk_file(WT_SESSION_IMPL *session, uint32_t queue_index, u_int *slotp)
 		/* Limit internal pages to 50% unless we get aggressive. */
 		if (WT_PAGE_IS_INTERNAL(page) &&
 		    !FLD_ISSET(cache->state, WT_EVICT_PASS_AGGRESSIVE) &&
-		    internal_pages >= (int)(evict - start) / 2)
+		    internal_pages >= (int)(end - start) / 2)
```

Comment:
@SunguckLee, I'm very concerned about making this change in general: I think there must be something going wrong elsewhere when WiredTiger is running your workload if there are no leaf pages in cache.

We could change it to this condition to make sure at least one internal page is considered:

```c
 		if (WT_PAGE_IS_INTERNAL(page) &&
  		    !FLD_ISSET(cache->state, WT_EVICT_PASS_AGGRESSIVE) &&
		    internal_pages > (int)(evict - start) / 2)
```

Can you check if that helps?